odoo/models.py

```diff
@@ -3671,12 +3691,10 @@ def _check_company(self, fnames=None):
             # with the company of the origin document, i.e. `self.account_id.company_id == self.company_id`
             for name in regular_fields:
                 corecord = record.sudo()[name]
-                # Special case with `res.users` since an user can belong to multiple companies.
-                if corecord._name == 'res.users' and corecord.company_ids:
-                    if not (company <= corecord.company_ids):
+                if corecord:
+                    domain = corecord.check_company_domain(company)
+                    if company and domain and not corecord.filtered_domain(domain):
                         inconsistencies.append((record, name, corecord))
```

Comment:
I don't remember why I put it like that. If the CI doesn't fail, good by me :+1:

Reply:
I seriously doubt it was the case since this test is failing and is explicitly setting a record linked to a company on a record without a company (`user_id` belongs to `company_main`, explicitly set in the setup)
https://github.com/odoo/odoo/blob/ac9162727cf282227ea5f6c5b8be9baaea9bc287/addons/sales_team/tests/common.py#L94-L99

Reply:
You're right, I changed the `_check_company_domain` on `res.users`